vars/patchDockerImage.groovy

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    String docker_image = "opensearchproject/${args.product}:${args.tag}"
+
+    sh"""
+    #!/bin/bash
+    set -e
+    set +x
+
+    docker pull ${docker_image}
+    """
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.version"}}' ${docker_image} > versionnumber"""
+
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.build-date"}}' ${docker_image} > time"""
+
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.description"}}' ${docker_image} > number"""
```

Comment:
Yeah but having all these statements in a single block isn't storing the output value in a variable. That's technically resulting an error.

Reply:
That doesn't sound right! This might be the culprit https://github.com/opensearch-project/opensearch-build-libraries/pull/268/files#diff-d42288632ebcbc435313fa808e87e49f58461165639396b6ec30cbb928596336R15-R17

I believe by default the shell is `zsh` and here you are running bash. Also those lines apply only to one command `docker pull ${docker_image}`. Try removing those and put it under default shell